### DISCUSSION TITLE
How to add function  at add button in repeater ?

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#6456] [Help]
How to add function  at add button in repeater ?

When user just clicked the Add Items in _Repeater Component_ , let say I want to check or validating the record, and display warning notification.

![image](https://user-images.githubusercontent.com/6214964/236977055-d1c6a55f-c9df-468a-b17b-e67d02534b5a.png)

Thank You

**Comment by jigujigu** [comment]
anyone can help me ?
thanks

**Comment by howdu** [comment]
Maybe there's a better way but you can overwrite the listeners event by creating a custom repeater class. 

[PR](https://github.com/filamentphp/filament/pull/6006) to make this easier got closed but Dan says 
> In v3 the event system is being replaced and you wont need an array for this anyway.

```php
class Repeater extends \Filament\Forms\Components\Repeater
{
    // Allow listener to be overwritten
    public function registerListener(string $name, Closure $callback): static
    {
        $this->listeners[$name] = [$callback];

        return $this;
    }
}
```

```php
Repeater::make()->registerListener('repeater::createItem',
    function ($component, string $statePath, ?array $data = null): void {
        if ($statePath !== $component->getStatePath()) {
            return;
        }

        $newUuid = (string) Str::uuid();

        $livewire = $component->getLivewire();
        data_set($livewire, "{$statePath}.{$newUuid}", []);

        $component->getChildComponentContainers()[$newUuid]->fill($data);

        $component->collapsed(false, shouldMakeComponentCollapsible: false);
        
        // Custom code
    }
)
```
